Examples/Filtering/DiffusionTensor3DReconstructionImageFilter.cxx

```diff
@@ -251,13 +251,19 @@ main(int argc, char * argv[])
           0.0) // this is a reference image
       {
         std::string fn("ReferenceImage%d.mhd");
-        snprintf(filename, sizeof(filename), fn.c_str(), referenceImageIndex);
+        ITK_GCC_PRAGMA_PUSH
+        ITK_GCC_SUPPRESS_Wformat_nonliteral snprintf(
+          filename, sizeof(filename), fn.c_str(), referenceImageIndex);
+        ITK_GCC_PRAGMA_POP
         ++referenceImageIndex;
```

Comment:
Instead of suppressing the warning, it seems to me that this code actually can be fixed, according to the warning message! The warning message says: _warning: format string is not a string literal_. Which suggests passing the string literal `"ReferenceImage%d.mhd"` directly to `snprintf`, without using an intermediate `fn` variable. Right?

Reply:
Thanks for the review @N-Dekker. Changed in 8984c2f.